src/parser.js

```diff
@@ -31,7 +31,14 @@ module.exports = function( configPath, defaultOpts ) {
 		disabledFeatures = opts.disabledFeatures ? opts.disabledFeatures.split( ',' ) : [];
 
 	configFiles.forEach( function( file ) {
-		Object.assign( data, getDataFromFile( file ) );
+		const fileData = getDataFromFile( file );
+		const features = Object.assign( data.features || {}, fileData.features || {} );
+
+		Object.assign( data, fileData );
+
+		if ( fileData.hasOwnProperty( 'features' ) ) {
+			data.features = features;
```

Comment:
Minor: we'll do this three time when all files are present and have the features object.

As a small optimization, we could move the features var definition before the foreEach and do the assignment to `data` after it.